openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java

```diff
@@ -756,30 +755,18 @@ private String extractCustomMediaType(FunctionDefinitionNode functionDefNode) {
             MetadataNode metadataNode = serviceDefNode.metadata().get();
             NodeList<AnnotationNode> annotations = metadataNode.annotations();
             if (!annotations.isEmpty()) {
-                mediaType = extractServiceAnnotationDetails(mediaType, annotations);
-            }
-        }
-        return mediaType;
-    }
-
-    private String extractServiceAnnotationDetails(String mediaType, NodeList<AnnotationNode> annotations) {
-        for (AnnotationNode annotation: annotations) {
-            Node annotReference = annotation.annotReference();
-            if (annotReference.toString().trim().equals("http:ServiceConfig") && annotation.annotValue().isPresent()) {
-                MappingConstructorExpressionNode listOfannotValue = annotation.annotValue().get();
-                for (MappingFieldNode field : listOfannotValue.fields()) {
-                    SpecificFieldNode media = (SpecificFieldNode) field;
-                    if ((media).fieldName().toString().trim().equals("mediaTypeSubtypePrefix")
-                            && media.valueExpr().isPresent()) {
-                        ExpressionNode expressionNode = media.valueExpr().get();
-                        mediaType = expressionNode.toString().trim().replaceAll("\"", "");
-                    }
+                Optional<String> value = ConverterCommonUtils.extractServiceAnnotationDetails(annotations,
+                        "http:ServiceConfig", "mediaTypeSubtypePrefix");
+                if (value.isPresent()) {
+                    mediaType = value.get();
```

Comment:
We modify this `extractCustomMediaType` method to return `Optional<String>` instead of empty strings.